kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/SolverServicesClient.java

```diff
@@ -41,6 +42,17 @@ void solvePlanningProblem(String containerId,
     void terminateSolverEarly(String containerId,
                               String solverId);
 
+    void addProblemFactChange(String containerId,
+                              String solverId,
+                              ProblemFactChange problemFactChange);
+
+    void addProblemFactChanges(String containerId,
+                              String solverId,
+                              List<ProblemFactChange> problemFactChange);
+
+    Boolean isEveryProblemFactChangeProcessed(String containerId,
+                                              String solverId);
```

Comment:
@ge0ffrey Yes, it doesn't look right to me as well. The problem is you can set different response handlers on the client. One of them org.kie.server.client.jms.FireAndForgetResponseHandler makes the client not wait for the server response. Instead the client just returns a null value.